disnake/types/interactions.py

```diff
@@ -189,7 +189,14 @@ class ComponentInteractionData(_ComponentInteractionDataOptional):
     component_type: ComponentType
 
 
-InteractionData = Union[ApplicationCommandInteractionData, ComponentInteractionData]
+class ModalInteractionData(TypedDict):
+    custom_id: str
+    components: List[ActionRow]
```

Comment:
Yeah, it would be nice to avoid using a lot of `Optional`s, but we would need to add a new `components.PartialInputText` to make that change relevant, otherwise, `components.ActionRow` would still use `components.InputText`.

Let's discuss this on Discord.